src/librustdoc/html/static/main.js

```diff
@@ -2830,25 +2821,27 @@ function defocusSearchBar() {
             return 0;
         });
         var savedCrate = getSettingValue("saved-filter-crate");
-        for (var i = 0; i < crates_text.length; ++i) {
-            var option = document.createElement("option");
-            option.value = crates_text[i];
-            option.innerText = crates_text[i];
+        var option, crate_text;
+        for (var i = 0, len = crates_text.length; i < len; ++i) {
+            option = document.createElement("option");
+            crate_text = crates_text[i];
+            option.value = crate_text;
+            option.innerText = crate_text;
             elem.appendChild(option);
             // Set the crate filter from saved storage, if the current page has the saved crate
             // filter.
             //
             // If not, ignore the crate filter -- we want to support filtering for crates on sites
             // like doc.rust-lang.org where the crates may differ from page to page while on the
             // same domain.
-            if (crates_text[i] === savedCrate) {
+            if (crate_text === savedCrate) {
                 elem.value = savedCrate;
             }
         }
         enableSearchInput();
     };
 
-    function buildHelperPopup() {
+    var buildHelperPopup = function() {
```

Comment:
Why this change?

Reply:
Because we call this function every time we want to show the help popup instead of creating the popup every time. However, since it only requires to be built once, we then set `buildHelperPopup` to an empty function so that calling it doesn't do anything anymore.